pkg/services/libraryelements/api.go

```diff
@@ -30,11 +30,31 @@ func (l *LibraryElementService) createHandler(c *models.ReqContext) response.Res
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
 
+	if cmd.FolderUID != nil {
+		if *cmd.FolderUID == "" {
+			cmd.FolderID = 0
+		} else {
+			folder, err := l.folderService.GetFolderByUID(c.Req.Context(), c.SignedInUser, c.OrgId, *cmd.FolderUID)
```

Comment:
Do we not want to do something here in case there's an error?

Reply:
this one indeed should be a bad request.